activemodel/lib/active_model/errors.rb

```diff
@@ -411,10 +411,9 @@ def full_messages_for(attribute)
     # * <tt>errors.format</tt>
     def full_message(attribute, message)
       return message if attribute == :base
-      attribute = attribute.to_s
+      attribute = attribute.to_s.remove(/\[\d\]/)
```

Comment:
I think the idea is that this was only done when `i18n_full_message` is set to true, otherwise it would be a change in behavior without deprecation.

Reply:
I actually tried extending `ActiveModel::Translation` so that it would respond to `i18n_scope`, but it didn't work.

If you think that's better I can dig deeper to make it work.